File: November 2024 Leetcode Solution/November-26.c
int findChampion(int n, int** edges, int edgesSize, int* edgesColSize) {
    int* indegree = (int*)calloc(n, sizeof(int));
    if (!indegree) return -1;

    for (int i = 0; i < edgesSize; i++) {
        indegree[edges[i][1]]++;
    }

    int champion = -1;
    for (int i = 0; i < n; i++) {
        if (indegree[i] == 0) {
            if (champion != -1) {
                free(indegree);
                return -1; 
            }
            champion = i;
        }
    }

    free(indegree);
    return champion;
}

File: November 2024 Leetcode Solution/November-26.rs
impl Solution {
    pub fn find_champion(n: i32, edges: Vec<Vec<i32>>) -> i32 {
        let mut indegree = std::collections::HashMap::new();

        for edge in edges {
            *indegree.entry(edge[1]).or_insert(0) += 1;
        }

        let mut champion = -1;

        for i in 0..n {
            if indegree.get(&i).copied().unwrap_or(0) == 0 {
                if champion != -1 {
                    return -1; 
                }
                champion = i;
            }
        }

        champion
    }
}
